shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -868,13 +877,27 @@ - (FlutterCompositor*)createFlutterCompositor {
 #pragma mark - Framework-internal methods
 
 - (void)addViewController:(FlutterViewController*)controller {
-  [self registerViewController:controller forId:kFlutterImplicitViewId];
+  NSCAssert(controller.engine == nil,
+            @"The FlutterViewController is unexpectedly attached to "
+            @"engine %@ before initialization.",
+            controller.engine);
```

Comment:
Why are we using NSCAssert here instead of NSAssert? NSCAssert should be reserved for plain C functions.

Reply:
Fixed! It was mistake due to the code movement.